src/Components/Gene/GeneArtworksContent.tsx

```diff
@@ -17,13 +18,22 @@ const SpinnerContainer = styled.div`
 const PageSize = 10
 
 export class GeneArtworksContent extends React.Component<Props, null> {
+  private finishedPaginatingWithError = false
+
   loadMoreArtworks() {
     const hasMore = this.props.filtered_artworks.artworks.pageInfo.hasNextPage
-    if (hasMore && !this.props.relay.isLoading()) {
+    const origLength = this.props.filtered_artworks.artworks.edges.length
+    if (hasMore && !this.props.relay.isLoading() && !this.finishedPaginatingWithError) {
       this.props.relay.loadMore(PageSize, error => {
         if (error) {
           console.error(error)
         }
+        const newLength = this.props.filtered_artworks.artworks.edges.length
```

Comment:
cc @joeyAghion get a load of this fanciness. Instead of endlessly fetching new pages, we'll error if we detect an anomaly- namely, counts indicating more pages available, yet pages coming back with less than what was requested. This is indicative of bad data...somewhere. So rather than endlessly spin/keep requesting higher and higher pages, we'll abort early, and `console.error` as well with some pertinent information (which should automatically make it to Sentry).

Reply:
How aggressive is this? I think some ES-backed endpoints _do_ post-filter unpublished works, and I wouldn't want that to crash front-ends.

Reply:
It's not going to crash...it just logs an error and will stop pagination.